src/ecs/rendering/systems.rs

```diff
@@ -12,25 +14,46 @@ pub struct RenderSystem<P> {
     pipe: P,
     #[derivative(Debug = "ignore")]
     renderer: Renderer,
+    reader_id: ReaderId,
 }
 
 impl<P> RenderSystem<P>
 where
     P: PolyPipeline,
 {
     /// Create a new render system
-    pub fn new(pipe: P, renderer: Renderer) -> Self {
-        Self { pipe, renderer }
+    pub fn new(pipe: P, renderer: Renderer, reader_id: ReaderId) -> Self {
+        Self {
+            pipe,
+            renderer,
+            reader_id,
+        }
     }
 }
 
 impl<'a, P> System<'a> for RenderSystem<P>
 where
     P: PolyPipeline,
 {
-    type SystemData = (Fetch<'a, Factory>, <P as PipelineData<'a>>::Data);
+    type SystemData = (
+        Fetch<'a, Factory>,
+        <P as PipelineData<'a>>::Data,
+        FetchMut<'a, EventHandler<WindowModifierEvent>>,
+    );
+
+    fn run(&mut self, (factory, data, events): Self::SystemData) {
+        //Read all window-modifying events
+        match events.read(&mut self.reader_id) {
+            Ok(EventReadData::Data(data)) => {
+                let mut win = self.renderer.window_mut();
+                for ev in data.collect::<Vec<_>>() {
+                    let closure = ev.modify;
+                    closure(win);
+                }
+            }
+            _ => {}
```

Comment:
Overflow variant might be good, I don't foresee it as a problem, but for completeness sake.